plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectLineRecordReader.java

```diff
@@ -110,7 +120,47 @@ public S3SelectLineRecordReader(
         closer.register(selectClient);
     }
 
-    public abstract SelectObjectContentRequest buildSelectObjectRequest(Properties schema, String query, Path path);
+    public abstract InputSerialization buildInputSerialization();
```

Comment:
Can be made `protected`.

Same for `buildOutputSerialization`, `shouldEnableScanRange`

Reply:
Updated